### ISSUE TITLE
"Move to new file" does not copy React import

### ISSUE THREAD

**Issue by OliverJAsh** [Microsoft/TypeScript#27937]
**TypeScript Version:**  3.1.3

**Search Terms:** react umd "move to new file" refactor

**Code**

```ts
// ./src/index.tsx
import * as React from 'react';

const MyComponent: React.SFC<{}> = props => <div />;

export default MyComponent;
```

``` json
{
    "compilerOptions": {
        "jsx": "react"
    },
    "include": ["./src/**/*"]
}
```

**Expected behavior:**
When copying the line `const MyComponent` and actioning the "Move to new file" command, the new file contents should be:

``` tsx
import * as React from 'react';
export const MyComponent: React.SFC<{}> = props => <div />;
```

**Actual behavior:**
The new file contents are:

``` tsx
export const MyComponent: React.SFC<{}> = props => <div />;
```

The `React` import is missing, which creates errors:

``` tsx
// React import is missing…
export const MyComponent: SFC<{}> = props =>
  // … which results in error: 'React' refers to a UMD global, but the current file is a module. Consider adding an import instead.
  <div />;
```

**Comment by amcasey** [comment]
I think we had some related issues in Organize Imports where he had to force React imports specifically (vs all UMD) to be treated as used because we were getting false negatives.

**Comment by amcasey** [comment]
Specifically, we probably need something like this

https://github.com/Microsoft/TypeScript/blob/aa3734c14834afb454c49bc956489980d990a765/src/services/organizeImports.ts#L137

here

https://github.com/Microsoft/TypeScript/blob/afbf89e3b3983fef4dfb5487556b374039e2744e/src/services/refactors/moveToNewFile.ts#L458

**Comment by amcasey** [comment]
Fixed by https://github.com/Microsoft/TypeScript/pull/29451.

**Comment by mizchi** [comment]
@amcasey 

Original issue is fixed, but I encountered next other reference error.

before

```tsx
// index.tsx
import React from 'react';
<div />
/* extract start */
function App() { return <div /> }
/* extract end */
```

after

```tsx
// index.tsx
<div /> // React reference error
```

**Comment by OliverJAsh** [comment]
@mizchi See https://github.com/Microsoft/TypeScript/issues/27937

**Comment by amcasey** [comment]
@mizchi - it looks like you might have linked to the current issue.  Did you mean to link to a different one?

**Comment by OliverJAsh** [comment]
Nope sorry I was confused. Ignore that comment

**Comment by amcasey** [comment]
@mizchi Sorry, I lost track of this because it's not an open issue.  Are you still having that problem?  It seems to work correctly for me.

```ts
/// <reference path='fourslash.ts' />

// @jsx: preserve
// @noLib: true
// @libFiles: react.d.ts,lib.d.ts,leftpad.d.ts

// @Filename: file.tsx
//// import React from 'react';
//// <div />
//// [|function App() { return <div /> }|]

verify.moveToNewFile({
    newFileContents: {
        "/tests/cases/fourslash/file.tsx":
`import React from 'react';
<div />
`,
        "/tests/cases/fourslash/App.tsx":
`import React from 'react';
function App() { return <div />; }
`,
    }
});
```

**Comment by OliverJAsh** [comment]
This still seems to be occurring, on both 3.5.2 and 3.6.2.

In fact, it's not specific to React. It's all namespace imports. I opened a new issue for that: https://github.com/microsoft/TypeScript/issues/33156.

**Comment by amcasey** [comment]
@OliverJAsh Thanks!
